_posts/2023-07-07-python-udf.md

```diff
@@ -0,0 +1,311 @@
+---
+
+layout: post
+title:  "From Waddle to Flying: Quickly expanding DuckDB's functionality with Scalar Python UDFs."
+author: Pedro Holanda, Thijs Bruineman and Phillip Cloud
+excerpt_separator: <!--more-->
+
+---
+
+<img src="images/pythonudf/bird-dance.gif"
+     alt="DuckDB-Waddle-fly"
+     width=100
+     />
+
+*TLDR: DuckDB now supports vectorized Scalar Python User Defined Functions (UDFs). By implementing Python UDFs, users can easily expand the functionality of DuckDB while taking advantage of DuckDB's fast execution model, SQL and data safety.*
+
+User Defined Functions (UDFs) enable users to extend the functionality of a Database Management System (DBMS) to perform domain-specific tasks that are not implemented as built-in functions. For instance, users who frequently need to export private data can benefit from an anonymization function that masks the local part of an email while preserving the domain. Ideally, this function would be executed directly in the DBMS. This approach offers several advantages:
+
+1) **Performance.** The function could be executed using the same execution model (e.g., streaming results, beyond-memory execution) of the DBMS, and without any unnecessary transformations.
```

Comment:
The term `beyond-memory execution` seems unique to DuckDB, see: https://www.google.com/search?q=%22beyond+memory+execution%22
`out-of-core execution` maybe?